File: src/main/java/com/baulsupp/oksocial/util/JsonUtil.java
package com.baulsupp.oksocial.util;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.IOException;
import java.util.Map;

public class JsonUtil {
  public static Map<String, Object> map(String content) throws IOException {
    ObjectMapper mapper = new ObjectMapper();
    return mapper.readValue(content, new TypeReference<Map<String, Object>>() {
    });
  }

  public static String toJson(Map<String, String> map) throws IOException {
    ObjectMapper mapper = new ObjectMapper();
    return mapper.writeValueAsString(map);
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/google/JsonUtil.kt
package com.baulsupp.oksocial.services.google

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.paramnames.ParameterNamesModule

object JsonUtil {
  fun map(content: String): Map<String, Any> {
    val mapper = jsonMapper()
    return mapper.readValue(content, object : TypeReference<Map<String, Any>>() {
    })
  }

  fun toJson(map: Map<String, String>): String {
    val mapper = jsonMapper()
    return mapper.writeValueAsString(map)
  }

  fun jsonMapper(): ObjectMapper {
    return ObjectMapper().registerModule(ParameterNamesModule())
      .registerModule(Jdk8Module())
      .registerModule(JavaTimeModule())
  }
}
